xword_dl/downloader/avcxdownloader.py

```diff
@@ -0,0 +1,227 @@
+
+        if not user_t or not token_t:
+            raise XWordDLException(
+                "No credentials provided or stored. Try running xword-dl avcx --authenticate."
+            )
+        else:
+            self.session.cookies.set("PA_AUS", user_t)
+            self.session.cookies.set("PA_ATOK", token_t)
+
+    @staticmethod
+    def matches_url(url_components):
+        return "avxwords.com" in url_components.netloc and (
+            "puzzles" in url_components.path or "download-puzzle" in url_components.path
+        )
+
+    def authenticate(self, username, password):
+        """Given an AVCX username and password, return username-based login token"""
+
+        try:
+            res = requests.post(
+                "https://avxwords.com/log-in/",
+                data={"email": username, "password": password},
+                allow_redirects=False,
+            )
+
+            res.raise_for_status()
+        except requests.HTTPError:
+            raise XWordDLException("Unable to authenticate.")
+
+        user_t = res.cookies.get("PA_AUS")
+        token_t = res.cookies.get("PA_ATOK")
+
+        if user_t and token_t:
+            update_config_file("avcx", {"AVCX-USER": user_t})
+            update_config_file("avcx", {"AVCX-TOKEN": token_t})
+            return (user_t, token_t)
+        else:
+            raise XWordDLException("No login token found in authentication response.")
+
+    def find_latest(self):
+        if self.puzzle_type is None:
+            raise XWordDLException(
+                "You must specify the AVCX variety to get the latest crossword."
+            )
+        return self.find_by_date()
+
+    def find_by_date(self, dt=None):
+        if self.puzzle_type is None:
+            raise XWordDLException(
+                "You must specify the AVCX variety to choose by date."
+            )
+        if dt is None:
+            year = datetime.now().strftime("%Y")
+        else:
+            year = dt.strftime("%Y")
+        res = self.session.get(f"https://avxwords.com/puzzles-by-year/?y={year}")
+        soup = bs(res.text, "lxml")
+        puzzle_list = next(
+            el.find_next("ul")
+            for el in soup.find_all("h3")
+            if el.string and self.puzzle_type in el.string
+        )
+
+        if dt is None:
+            latest_url = puzzle_list.find("li", class_="row").find("a").get("href")
+            return urljoin(self.base_url, latest_url)
+
+        for row in puzzle_list.find_all("li", class_="row"):
+            date = row.find("span", class_="puzzle-date").text.strip()
+            row_dt = datetime.strptime(date, "%b %d, %Y")
+            if row_dt.date() == dt.date():
+                url = row.find("a").get("href")
+                return urljoin(self.base_url, url)
+
+        raise XWordDLException("Could not find puzzle for date.")
+
+    def find_solver(self, url):
+        if "puzzles" in url:
+            url = url.removesuffix("/")
```

Comment:
This is 3.9+ and 3.8 is not quite EOL yet but I'm okay with that

Reply:
Yep, normally I'd be all for backward compatibility, but 3.8 will probably be EOL before the next release of xword-dl, and at this point even Debian oldstable has 3.9. Still, willing to to change it if you'd prefer.